exonum/src/blockchain/mod.rs

```diff
@@ -386,23 +386,23 @@ impl Blockchain {
     }
 
     /// Saves raw message to the consensus messages cache
-    pub fn save_message(&mut self, raw: &RawMessage) {
+    pub fn save_message(&mut self, raw: &RawMessage) -> Result<(), Error> {
         let mut fork = self.fork();
 
         {
             let mut schema = Schema::new(&mut fork);
             schema.consensus_messages_cache_mut().push(raw.clone());
         }
 
-        // apply changes
-        self.merge(fork.into_patch()).unwrap();
+        self.merge(fork.into_patch())?;
+        Ok(())
```

Comment:
This can be rewritten the following way:
```rust
self.merge(fork.into_patch())
```

Reply:
Indeed. Somehow I mixed it with a different method and thought self.merge() returns Result<T, Error> instead of Result<(), Error>.